### DISCUSSION TITLE
Corporate Sponsor?

### DISCUSSION THREAD

**Discussion by ad3154** [Open-Agriculture/AgIsoStack-rs#1] [Announcements]
Hello members of Open-Agriculture and those who use AgIsoStack!

As some of you may know, I started developing AgIsoStack++, this repo's more complete counterpart, just over a year ago because I saw a need in the open source community for a free, high quality ISOBUS CAN stack in C++.
Since then, we've added numerous features, learned a lot, grown as a community, and collectively decided we also want a Rust CAN stack 🦀.

As some of you might know if you've checked my profile here on GitHub, I work for a company that's a member of AEF, Raven Industries (a part of CNHi).

Raven has been developing ISOBUS ECUs for something like twenty years now including field computers, rate controllers, guidance and steering products, and more.

Here at Raven we periodically have time to work on "innovation" during something called our "Innovation Sprints" where our usual development work is deprioritized for engineers to do cool things!

This year, some of us want to work on an open source CAN stack, particularly in Rust. Some of us develop in Rust, and others are learning.

## The Big Announcement

I'm happy to say Raven has agreed to sponsor AgIsoStack-rs by contributing to it if the community here is on-board.

### What This Means

If the community agrees, this means that multiple Raven engineers will (when time allows - such as during innovation sprints) contribute code to help develop AgIsoStack-rs under the MIT license, which will ultimately help us make progress faster. It also means that the code we work on here may be more likely to be used in real Raven products some day (though, no promises). 

This also means that we may want to quickly prioritize discussions about design and structure of this project to align on what exactly we're creating.

This sponsorship would not affect AgIsoStack++.

### Caveats

This sponsorship also comes with a requirement from Raven, and that is that Raven would want to own the copyright to the code in the repo.
This would not change the license, it would remain MIT. 

This would generally involve us creating a `COPYRIGHT` file in the root of our repo with the following language:

```
Copyright Ownership: Contributing to this software is pursuant to the license agreement within GitHub available on GitHub’s website, https://docs.github.com/en/site-policy/github-terms/github-terms-of-service#d-user-generated-content. This additional provision is meant to clarify, and not replace or alter the terms on GitHub’s terms of service. Contribution to the code within this content started by Raven Industries, Inc. and its affiliated companies (“Raven”) vests in Raven all copyrights in content, subject to the non-exclusive license granted by GitHub in the GitHub Terms and Conditions. By contributing to the code, You agree Raven owns the copyright in the code as stated in this provision.
```

Additionally, we would need to have a `// Copyright Raven Industries, inc.` somewhere at the top of source files.

The Open-Agriculture organization would continue being the home for the repo for the forseeable future.

Personally, I think this is a fairly minor cost to have more contributors and official support from Raven, but I could see this not being loved from an open source purist perspective.

### Conclusion

I want to emphasize that this is a group choice. Please comment on this announcement if you have questions and I will work with my Raven peers to answer you. Please also comment your thoughts. If everyone is OK with it, then I will invite some additional members to the organization as needed, and one of us at Raven will open PRs for our changes, at which time of course your comments will also be welcome as usual.

I think this could be a great opportunity for us.

Thanks,

Adrian

**Comment by Thom-de-Jong** [comment]
Wow that is a great opportunity for the growth of this repo!

Personally, while I think it is not great to give the copyright to Raven. I can understand it from there point of view.
As long as the code stays MIT Licensed and I can use it within my company, I'm ok with that tradeoff.

I have not had the time yet to produce any production grade code but we should indeed start discussing the design of the project.

Once again, great opportunity :+1:

**Comment by JannesBrands** [comment]
Generally I agree with @Thom-de-Jong!
On the other hand, it also needs the support of a company, since it is difficult to make progress due to the very small community. But I am really suprised and happy to hear Raven is open to contribute and share it for the public community. There are some ISOBUS frameworks like those of CCI and John Deere circulating, which are either not freely accessible or simply financially for smaller companies or for the private purpose does not make sense.
I am curious to hear more opinions and perhaps more exchange with Raven.

**Reply by JannesBrands** [reply]
on the other hand, by exempting the framework Raven should also consider that through this growing community will also contribute more and also receive more feedback regarding bugs, which also has a business advantage. There should be some more exchange to discuss these points. Even the openness for the project is still remarkable.

**Reply by ad3154** [reply]
I am not sure I follow what you mean when you say `exempting the framework`? Can you elaborate?

**Reply by JannesBrands** [reply]
Sorry, maybe the word exempting caused a misunderstanding. I had translated this from the German. I rather meant that raven gets the copyright.

**Comment by Combatjuan** [comment]
Hello all.  Charles here.  I currently manage the embedded software teams at Raven and am passionate about the value of Open Source so I just thought I'd offer my support of and perspective on this project.

I see high quality and compatible implementations of agriculture standards as tools that farmers, implement manufacturers, hobbiest hackers, and big OEMs alike can all use for universal benefit.  Good ISO CAN stacks are a huge enabler for us all.  And one written in Rust is something that I find personally exciting.

Raven, and it's parent company Case New Holland has been very supportive of these kinds of efforts and I hope to be able to promote more of them in the future.

Getting a bunch of eyes on and users of any software is the best way to increase its quality and arrive at a solid implementation.  I am looking forward to the day where we can adopt this and other projects like it into our products.  That's a ways off for this project (being in such a nascent stage) but you'd be hard pressed to find a better steward than Adrian.

Thanks to all of you for your thoughts and contributions!

**Comment by ad3154** [comment]
@GwnDaan I am particularly interested in your opinion as well

**Reply by GwnDaan** [reply]
I think it's a great compromise. Getting a company involved should offer lots of stability. Though I have some questions;
1. Does this tradeoff mean they have exclusive rights to the code written? E.g. the license could be changed  from MIT to another at any time?
 2. Are we still having discussions about the project on Github? 

Nevertheless great opportunity!

**Reply by ad3154** [reply]
Good questions, my understanding is as follows (and @Combatjuan feel free to chime in if you disagree):

1. Raven owning the copyright does not mean they have exclusive rights to the code written, since the code would be MIT licensed. Anyone acquiring a copy of the software would have rights to use it as they see fit within the terms of the MIT license.
2. Raven could as the copyright owner _in theory_ change the license if they want in the future (though I seriously doubt they would). Even then, rights given to you or other people who acquire the software by the MIT license cannot just be revoked at will by Raven. So if raven did change the license, that would essentially lead to a "fork" of the repo. I will reiterate that this seems very unlikely to me. The only licenses even mentioned to me internally as acceptable for the repo were MIT or BSD, so I think you're in pretty safe hands.
4. Yes we'll still have our discussions, PRs, pipelines, issue tracking, etc on GitHub. It'll still be a community effort.

**Reply by Combatjuan** [reply]
@ad3154  - My interpretation is identical to yours on the 3 points.

It's theoretically possible that a future Raven regime could choose to create a commercial fork (e.g. that has support for some compliance framework that businesses would pay for or that specifically targets large commercial library consumers).  Or quite alternatively maybe Richard Stallman will be elected to the head of the board and insist all Raven software shift to a more copyleft license such as GPLv3.  In either case, the community could fork or continue to maintain the MIT version.

I agree that those possibilities are highly unlikely and also very low risk.
